src/modules/tests/tests.analytics.effects.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+import { AnalyticsProvider } from '../../providers/analytics/analytics';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { Store, select } from '@ngrx/store';
+import { StoreModel } from '../../shared/models/store.model';
+import { SET_TEST_STATUS_SUBMITTED, SetTestStatusSubmitted } from './test-status/test-status.actions';
+import { withLatestFrom, tap } from 'rxjs/operators';
+import { getTests } from './tests.reducer';
+import {
+  AnalyticsEventCategories,
+  AnalyticsEvents,
+  AnalyticsDimensionIndices,
+} from '../../providers/analytics/analytics.model';
+import { getTestById, isPassed } from './tests.selector';
+
+@Injectable()
+export class TestsAnalyticsEffects {
+  constructor(
+    private analytics: AnalyticsProvider,
+    private actions$: Actions,
+    private store$: Store<StoreModel>,
+  ) {
+    this.analytics.initialiseAnalytics()
+      .then(() => {})
+      .catch(() => {
+        console.log('error initialising analytics');
+      },
+    );
+  }
+
+  @Effect()
+  setTestStatusSubmittedEffect$ = this.actions$.pipe(
+    ofType(SET_TEST_STATUS_SUBMITTED),
+    withLatestFrom(
+      this.store$.pipe(
+        select(getTests),
+      ),
+    ),
+    tap(([action, tests]) => {
+      const setTestStatusSubmittedAction = action as SetTestStatusSubmitted;
+      const test = getTestById(tests, setTestStatusSubmittedAction.slotId);
+      const isTestPassed = isPassed(test);
+      const journalDataOfTest = test.journalData;
+
+      this.analytics.logEvent(
+        AnalyticsEventCategories.POST_TEST,
+        AnalyticsEvents.SUBMIT_TEST,
+        isTestPassed ? 'pass' : 'fail',
+      );
+
+      this.analytics.addCustomDimension(
+        AnalyticsDimensionIndices.TEST_ID, journalDataOfTest.testSlotAttributes.slotId.toString());
```

Comment:
Could you use the `getCurrentTestSlotId` selector for this?

Reply:
Ah maybe not... that selects `tests.currentTest.slotId`

Reply:
Oh, you could just select `slotId` earlier and then reuse
`const slotId = getCurrentTestSlotId(tests);`

Reply:
The only thing with the `getCurrentTestSlotId` function is that it always gets the current test slot id. And there is a chance here that we are not setting the status of the current test, but an older test which failed to get submitted earlier. 
And the most easiest way is to go check what's in `journalData` and set that `slotId`.